src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -36,6 +34,8 @@
 @Component
 public class DatasetRequestValidator implements Validator {
 
+    private final static ImmutableSet VALID_DATATYPES = ImmutableSet.of("fileref", "int64", "float64", "string", "date");
```

Comment:
I'm surprised that this list doesn't already exist somewhere. Not sure if this is helpful, but here's one place I found a similar list of types. https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java#L1429